tests/component/test_stream_readers_di.py

```diff
@@ -179,12 +179,12 @@ def _bool_array_to_int(bool_array: numpy.typing.NDArray[numpy.bool_]) -> int:
     return result
 
 
-_DType = TypeVar("_DType", bound=numpy.typing.DTypeLike)
+_D = TypeVar("_D", bound=numpy.generic, covariant=True)
```

Comment:
@zhindes It turns out numpy.generic was right after all, but we want a type var to make the return type preserve the parameter type.

This also implicitly relies on the fact that the DAQmx read methods are untyped (Any).